server/tests/src/test/java/org/infinispan/server/resilience/ResilienceMaxRetryIT.java

```diff
@@ -34,78 +41,65 @@
 
-      // 4 -> kill localhost:port1
-      InetAddress killedServer = SERVERS.getServerDriver().getServerAddress(0);
-      SERVERS.getServerDriver().kill(0);
+      // Stop server0
+      InetSocketAddress stoppedServerAddress = SERVERS.getServerDriver().getServerSocket(0, SERVERS.getTestServer().getDefaultPortNumber());
+      SERVERS.getServerDriver().stop(0);
 
-      // 5 -> do more operations, check that it's only connected to port2 and port3
-      // thinking one failure would be enough, but the topology is actually updated only when the client connects to one of the live servers
-      // you could have the first 2 connections to port1 and they'd both fail
-      // more than 2 is highly unlikely, but it might happen because the probability is not 0
-      for (int i = 0; i < 100; i++) {
-         String random = UUID.randomUUID().toString();
+      // Execute cache operations so the client connects to server1 or server2 and receives a topology update
+      // The client keeps trying to connect to failed servers for each new operation,
+      // so the number of operations we need depends on which server owns each key
+      byte[] cacheNameBytes = cache.getName().getBytes();
+      for (int i = 0; i < 10; i++) {
          try {
-            cache.put(random, random);
+            cache.get(ThreadLocalRandom.current().nextInt());
+            break;
          } catch (TransportException e) {
-            // if you do 10 operations after you kill port1, the first 0 or 1 or 2 operations might fail.
-            // But if the 3rd operation is successful, then operations 4-10 should be successful as well
-            assert i == 0 || i == 1 || i == 2;
-            // assert that the failed server is the one that we killed
-            assert e.getMessage().contains(killedServer.toString());
+            // Assert that the failed server is the one that we killed
+            assert e.getMessage().contains(stoppedServerAddress.toString()) : e.getMessage();
          }
       }
 
-      // double check that the killed server was properly removed from the list
-      Collection<InetSocketAddress> currentServers = cache.getRemoteCacheManager().getChannelFactory().getServers(cache.getName().getBytes());
-      assert currentServers.size() == 2;
+      // Check that the stopped server was properly removed from the list
+      Collection<InetSocketAddress> currentServers = cache.getRemoteCacheManager().getChannelFactory().getServers(cacheNameBytes);
+      assertEquals(2, currentServers.size());
       for (InetSocketAddress currentServer : currentServers) {
-         if (currentServer.getHostString().equals(killedServer.getHostAddress())) {
-            throw new IllegalStateException("The removed server should not be present in the list");
+         if (currentServer.equals(stoppedServerAddress)) {
+            throw new IllegalStateException(stoppedServerAddress + " should not be present in the list: " + currentServers);
          }
       }
```

Comment:
`assertFalse(currentServers.contains(stoppedServerAddress));`?

Reply:
I replaced it with a positive assert